src/pages/HabitTracker/Habits.tsx

```diff
@@ -144,29 +311,121 @@ export function Habits() {
     }
   };
 
+  // Handler for logging habit activity
+  const handleLog = (habitName: string, selectedDate: Date) => {
+    manageLogging(habitName, selectedDate, logging, setLogging, "add");
+  };
+
+  // Handler for removing a log
+  const handleRemoveLog = (habitName: string, selectedDate: Date) => {
+    manageLogging(habitName, selectedDate, logging, setLogging, "remove");
+  };
+
+  // Determine if the Remove Log button should be disabled
+  const isRemoveLogDisabled = (habitName: string, selectedDate: Date) => {
+    const year = selectedDate.getFullYear();
+    const month = selectedDate
+      .toLocaleString("default", { month: "long" })
+      .toLowerCase();
+    const day = selectedDate.getDate();
+
+    return !(
+      logging[habitName] &&
+      logging[habitName][year] &&
+      logging[habitName][year][month] &&
+      logging[habitName][year][month][day] > 0
+    );
+  };
+
   // Check if the Save button should be enabled
   const isSaveDisabled = () =>
     !newHabit.name ||
     newHabit.count <= 0 ||
     (newHabit.name === originalHabit.name &&
       newHabit.count === originalHabit.count);
 
+  // Function to render the habit list
+  const renderHabits = () => {
+    return habits.map((habit, index) => {
+      const year = selectedDate.getFullYear();
+      const month = selectedDate
+        .toLocaleString("default", { month: "long" })
+        .toLowerCase();
+      const day = selectedDate.getDate();
+
+      const logCount =
+        logging[habit.name] &&
+        logging[habit.name][year] &&
+        logging[habit.name][year][month] &&
+        logging[habit.name][year][month][day]
+          ? logging[habit.name][year][month][day]
+          : 0;
```

Comment:
I don't like these checks. There are a few things we can do to avoid this:

1. Add type checking to the `logging` state, which will ensure that none of these objects will be empty
2. Read about optional chaining. That will avoid having to make these checks. It would look something like:
`logging?.name?.year?.month?.day`